optimade/models/structures.py

```diff
@@ -35,6 +35,24 @@ class Periodicity(IntEnum):
     PERIODIC = 1
 
 
+class SupportLevel(Enum):
+    MUST = "must"
+    SHOULD = "should"
+    OPTIONAL = "optional"
+
+
+def OptimadeField(
+    *args,
+    queryable: SupportLevel = SupportLevel.MUST,
+    support: SupportLevel = SupportLevel.MUST,
+    **kwargs,
+):
+    kwargs.update(
+        {"x-optimade-queryable": queryable, "x-optimade-support-level": support}
+    )
+    return Field(*args, **kwargs)
+
+
```

Comment:
If we think these are a reasonable idea, then we can move them out of structures and probably add more Support Levels (e.g. RECOMMENDED, MAY)

Reply:
I agree 😃 It would good to get it into a more general place, e.g., the models.utils?